src/external/bincode_support.rs

```diff
@@ -0,0 +1,259 @@
+// Copyright 2013-2014 The Rust Project Developers.
+// Copyright 2018 The Uuid Project Developers.
+//
+// See the COPYRIGHT file at the top-level directory of this distribution.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use crate::{
+    convert::TryFrom,
+    fmt::{Braced, Hyphenated, Simple, Urn},
+    non_nil::NonNilUuid,
+    Uuid,
+};
+use bincode::{
+    config::{Config, IntEncoding},
+    de::{BorrowDecoder, Decoder},
+    enc::Encoder,
+    error::{DecodeError, EncodeError},
+    BorrowDecode, Decode, Encode,
+};
+use std::string::ToString;
+
+impl Encode for Uuid {
+    fn encode<E: Encoder>(&self, encoder: &mut E) -> Result<(), EncodeError> {
+        if encoder.config().int_encoding() == IntEncoding::Fixed {
+            usize::encode(&16, encoder)?;
+        } else {
+            u8::encode(&16, encoder)?;
+        }
+
+        <&[u8; 16]>::encode(&self.as_bytes(), encoder)?;
+
+        Ok(())
+    }
+}
+
+impl Encode for NonNilUuid {
+    fn encode<E: Encoder>(&self, encoder: &mut E) -> Result<(), EncodeError> {
+        self.get().encode(encoder)
+    }
+}
+
+impl Encode for Hyphenated {
+    fn encode<E: Encoder>(&self, encoder: &mut E) -> Result<(), EncodeError> {
+        self.as_uuid().encode(encoder)
+    }
+}
+
+impl Encode for Simple {
+    fn encode<E: Encoder>(&self, encoder: &mut E) -> Result<(), EncodeError> {
+        self.as_uuid().encode(encoder)
+    }
+}
+
+impl Encode for Urn {
+    fn encode<E: Encoder>(&self, encoder: &mut E) -> Result<(), EncodeError> {
+        self.as_uuid().encode(encoder)
+    }
+}
+
+impl Encode for Braced {
+    fn encode<E: Encoder>(&self, encoder: &mut E) -> Result<(), EncodeError> {
+        self.as_uuid().encode(encoder)
+    }
+}
```

Comment:
Wouldn't these differ from the `serde` impl?

`serde` would output strings for these, therefore I believe bincode v1 encodes a string rather than the bytes.

Do we care, though?

Reply:
Ah well spotted! We definitely should maintain compatibility on these adapter types too